packages/microblog/src/client.ts

```diff
@@ -0,0 +1,284 @@
+import { AdxUri } from '@adxp/common'
+import { AdxClient, AdxPdsClient, AdxRepoClient } from '@adxp/api'
+import {
+  Post,
+  Profile,
+  Like,
+  Repost,
+  Follow,
+  Badge,
+  FeedView,
+  LikedByView,
+  PostThreadView,
+  ProfileView,
+  RepostedByView,
+  UserFollowersView,
+  UserFollowsView,
+} from '@adxp/microblog'
+import { schemas } from './defs'
+
+export class MicroblogClient {
+  public client: AdxClient
+  public pds: AdxPdsClient
+  public repo: AdxRepoClient
+
+  constructor(public pdsUrl: string, public did: string) {
+    this.client = new AdxClient({
+      pds: pdsUrl,
+      schemas,
+    })
+    this.pds = this.client.mainPds
+    this.repo = this.client.repo(did)
+  }
+
+  config() {
+    return {
+      headers: {
+        Authorization: this.did,
+      },
+    }
+  }
+
+  async register(username: string) {
+    await this.pds.registerRepo({ username, did: this.did })
+  }
+
+  async createProfile(
+    displayName: string,
+    description?: string,
+    hardcode: Partial<Profile.Record> = {},
+  ): Promise<AdxUri> {
+    const uri = await this.repo
+      .collection('bsky/profile')
+      .create('blueskyweb.xyz:Profile', {
+        $type: 'blueskyweb.xyz:Profile',
+        displayName,
+        description,
+        ...hardcode,
+      })
+    return uri
+  }
+
+  async createPost(
+    text: string,
+    entities?: Post.Entity[],
+    hardcode: Partial<Post.Record> = {},
```

Comment:
What's this `hardcode` pattern?

Reply:
There could probably be another term for it. But it essentially lets you "hardcode" certain attributes on records that aren't covered by the semantic interface. So for instance, providing a different `createdAt`.

I'm not totally sure it's necessary. I added it in when I was going to create records with deterministic timestamps but then I ended up not pursuing that idea.

Reply:
Yeah makes sense. Okay up to you.

Reply:
decided to ditch this pattern. I think it adds unneeded complexity & if you need to drop down into hardcoding certain record values, you can always do it by just using the AdxClient